src/services/accessValidation.js

```diff
@@ -19,23 +19,35 @@ const logContext = {
  * @returns {Promise<any>} - returns a promise
  */
 export default function findOrCreateUser(data) {
-  return sequelize.transaction((transaction) => User.findOrCreate({
+  return sequelize.transaction((transaction) => User.findOne({
     where: {
       hsesUserId: data.hsesUserId,
     },
+    transaction,
+  }).then((userFoundByHsesUserId) => User.findOrCreate({
```

Comment:
I don't like calling `findOrCreate` no matter what. I think we should call it only if `userFoundByHsesUserId` is `null`. That would also let the `where` be simplified a bit to just `where: { hsesUsername: data.hsesUsername }`

I'm worried about potentially unusual circumstances where the ID and username both exist but on different records. Which one would then be logged in?

Reply:
Ok, I changed it. Good catch, if that's a possibility.

Reply:
Honestly, it probably shouldn't be a possibility, but...

1) If that's wrong this is safer now
2) Now we only need one `SELECT` to login a user in production instead of 2.